build/commands/lib/config.js

```diff
@@ -52,13 +52,28 @@ var packageConfigBraveCore = function (key) {
   return packageConfig(key, path.join(rootDir, 'src', 'brave'))
 }
 
-const getNPMConfig = (key) => {
+const getNPMConfig = (key, default_value = undefined) => {
   if (!NpmConfig) {
-    const list = run(npmCommand, ['config', 'list', '--json', '--userconfig=' + path.join(rootDir, '.npmrc')])
+    const list = run(npmCommand, ['config', 'list', '--json'], { cwd: rootDir })
     NpmConfig = JSON.parse(list.stdout.toString())
   }
 
-  return NpmConfig[key.join('-').replace(/_/g, '-')] || packageConfigBraveCore(key) || packageConfig(key)
+  // NpmConfig has the multiple copy of the same variable: one from .npmrc
+  // (that we want to) and one from the environment.
+  // https://docs.npmjs.com/cli/v7/using-npm/config#environment-variables
+  const npmConfigValue = NpmConfig[key.join('_')]
+  if (npmConfigValue !== undefined)
+    return npmConfigValue
+
+  const packageConfigBraveCoreValue = packageConfigBraveCore(key)
+  if (packageConfigBraveCoreValue !== undefined)
+    return packageConfigBraveCoreValue
+
+  const packageConfigValue = packageConfig(key)
```

Comment:
I don't think there's anything we need to read from brave-browser package.json anymore?

Reply:
shouldn't be

Reply:
Ah, we can't remove this code until https://github.com/brave/brave-core/pull/11543/files is merged
`  this.braveCoreRepo = getNPMConfig(['projects', 'brave-core', 'repository', 'url'])`

Reply:
Hopefully today.